yogstation/code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -0,0 +1,26 @@
+/datum/species/moth
+	var/MothLightRange = 15 //In what radius shall moths look for lights to fly at?
+	var/MothLightProb = 2 //Don't want them doing it tooooo often..Unless an admin changes it :^) So change this back to 5 after testing
+
+/datum/species/moth/spec_life(mob/living/carbon/human/H)
+	. = ..()
+	if(H.stat != DEAD) //Suffer not a dead moth to move
+		MothLightProb = initial(MothLightProb)
+		var/retarded = H.getBrainLoss()
+		MothLightProb += retarded*0.3 //So there's always a slight chance.
+		var/list/LightsIWantToFlyAt = list()
+		if(prob(MothLightProb)) //Moth people now have a random urge to fly violently into lights
+			for(var/obj/O in orange(H,MothLightRange))
+				if(O.light_sources)
+					for(var/datum/light_source/S in O.light_sources)
+						if(istype(S, /datum/light_source))
+							if(S.light_power > 0)
+								for(var/i = 1 to round(S.light_power)) //Weight the list according to light power
+									LightsIWantToFlyAt += S.source_atom
+			var/atom/MothWantsToFlyAtThisOne = pick(LightsIWantToFlyAt)
```

Comment:
there's a much better way to do this for loop and pick.

>LightsIWantToFlyAt += S.source_atom
>LightsIWantToFlyAt[S.source_atom] = S.light_power
>MothWantsToFlyAtThisOne = pickweight(lightsIWantToFlyAt)

Reply:
spaghetti with extra ketchup

Reply:
it only works on dictionaries and i remember tg started debatting about why it exists when i pointed it out